src/stage1/ir.cpp

```diff
@@ -29544,6 +29544,23 @@ static IrInstGen *ir_analyze_instruction_check_switch_prongs(IrAnalyze *ira,
                     }
                 }
             }
+        } else {
+            if (!type_is_global_error_set(switch_type)) {
+                bool needs_else = false;
+                for (uint32_t i = 0; i < switch_type->data.error_set.err_count; i += 1) {
```

Comment:
Isn't this a more expensive version of `const bool needs_else = instruction->range_count != switch_type->data.error_set.err_count` ?
By the time you reach this line the compiler already knows all the values are correctly typed and the prong list contains no duplicates.